db/migrate/20180723105200_add_docker_manifest_platform.rb

```diff
@@ -0,0 +1,51 @@
+class AddDockerManifestPlatform < ActiveRecord::Migration[5.1]
+  def up
+    create_table :katello_docker_manifest_platforms do |t|
+      t.string :arch, :limit => 255
+      t.string :os, :limit => 255
+      t.timestamps
+    end
+
+    create_table :katello_docker_manifest_manifest_platforms do |t|
+      t.references :docker_manifest, :null => false, :index => false
+      t.references :docker_manifest_platform, :null => false, :index => false
+      t.timestamps
+    end
+
+    create_table :katello_docker_manifest_list_manifest_platforms do |t|
```

Comment:
Why do we need these join tables? Let's just use `has_and_belongs_to_many`, unless you think there will be validations or other logic added to these models. See the [rails guide](https://guides.rubyonrails.org/association_basics.html#choosing-between-has-many-through-and-has-and-belongs-to-many) about when to use what.

Reply:
@thomasmckay This change would simplify and maybe make [this comment](https://github.com/Katello/katello/pull/7581#discussion_r209966134) moot.